File: AndroidStudioProjects/034ContextualMenu/app/src/main/java/com/example/a034contextualmenu/MainActivity.java
package com.example.a034contextualmenu;

import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.view.ActionMode;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.ImageView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    ActionMode mActionMode;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        //onLongClickListener en el imageView para el menu contextual
        ImageView imageView = findViewById(R.id.imageView);
        imageView.setOnLongClickListener(new View.OnLongClickListener() {
            @SuppressLint("SuspiciousIndentation")
            @Override
            public boolean onLongClick(View view) {
                if(mActionMode != null)
                return false;
                mActionMode = startActionMode(mActionModeCallback);
                return true;
            }
        });

    }

    private ActionMode.Callback mActionModeCallback = new ActionMode.Callback() {
        @Override
        public boolean onCreateActionMode(ActionMode mode, Menu menu) {
            mode.getMenuInflater().inflate(R.menu.context_menu, menu);
            return true;
        }

        @Override
        public boolean onPrepareActionMode(ActionMode mode, Menu menu) {
            return false;
        }

        @SuppressLint("NonConstantResourceId")
        @Override
        public boolean onActionItemClicked(ActionMode mode, MenuItem item) {
            int id = item.getItemId();
            if(id==R.id.menu_cast) {
                Toast.makeText(MainActivity.this, "Cast", Toast.LENGTH_SHORT).show();
                mode.finish();
                return true;
            }else if(id==R.id.menu_print){
                    Toast.makeText(MainActivity.this,"Print" ,Toast.LENGTH_SHORT).show();
                    mode.finish();
                    return true;
            }
            return false;
        }

        @Override
        public void onDestroyActionMode(ActionMode mode) {
            mActionMode = null;
        }
    };
}

File: AndroidStudioProjects/contextual-action-mode/FloatingActionMode/src/main/java/com/develou/floating_actionmode/MainActivity.kt
package com.develou.floating_actionmode

import android.graphics.Rect
import android.os.Build
import android.os.Bundle
import android.view.ActionMode
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.Toast
import androidx.annotation.RequiresApi
import androidx.appcompat.app.AppCompatActivity

@RequiresApi(Build.VERSION_CODES.M)
class MainActivity : AppCompatActivity() {
    private var actionMode: ActionMode? = null
    private lateinit var image: View

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<View>(android.R.id.content).setOnClickListener {
            actionMode?.finish()
        }

        image = findViewById(R.id.image_container)
        image.setOnLongClickListener { view ->
            when (actionMode) {
                null -> {
                    actionMode = view.startActionMode(floatingActionMode, ActionMode.TYPE_FLOATING)
                    view.isActivated = true
                    true
                }
                else -> false
            }
        }
    }

    private val floatingActionMode = object : ActionMode.Callback2() {
        override fun onCreateActionMode(mode: ActionMode, menu: Menu): Boolean {
            menuInflater.inflate(R.menu.floating_cab_menu, menu)
            return true
        }

        override fun onPrepareActionMode(mode: ActionMode, menu: Menu): Boolean {
            return false
        }

        override fun onActionItemClicked(mode: ActionMode, item: MenuItem): Boolean {
            return when (item.itemId) {
                R.id.upload, R.id.scale, R.id.resize, R.id.crop -> {
                    Toast.makeText(this@MainActivity, "Acción: ${item.title}", Toast.LENGTH_SHORT)
                        .show()
                    mode.finish()
                    true
                }
                else -> false
            }
        }

        override fun onDestroyActionMode(mode: ActionMode) {
            actionMode = null
            image.isActivated = false
        }

        override fun onGetContentRect(mode: ActionMode?, view: View?, outRect: Rect?) {
            super.onGetContentRect(mode, view, outRect)
            // TODO: aquí usarías tus coordenadas con outRect
            // ej: outRect.set(16,16,16,16)
        }
    }
}